src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -79,54 +79,49 @@ public GitService() {
      * @throws IOException
      * @throws InterruptedException
      */
-    public Repository getOrCheckoutRepository(Participation participation) throws IOException, InterruptedException {
+    public Repository getOrCheckoutRepository(Participation participation) throws IOException, InterruptedException, GitAPIException {
         URL repoUrl = participation.getRepositoryUrlAsUrl();
-        Repository repository = getOrCheckoutRepository(repoUrl);
+        Repository repository = getOrCheckoutRepository(repoUrl, true);
         repository.setParticipation(participation);
         return repository;
     }
 
     /**
      * Get the local repository for a given remote repository URL. If the local repo does not exist yet, it will be checked out.
      *
-     * @param repoUrl The remote repository.
+     * @param repoUrl   The remote repository.
+     * @param pullOnGet Pull from the remote on the checked out repository, if it does not need to be cloned.
      * @return
      * @throws IOException
      * @throws InterruptedException
      */
-    public Repository getOrCheckoutRepository(URL repoUrl) throws IOException, InterruptedException {
+    public Repository getOrCheckoutRepository(URL repoUrl, boolean pullOnGet) throws IOException, InterruptedException, GitAPIException {
 
         Path localPath = new java.io.File(REPO_CLONE_PATH + folderNameForRepositoryUrl(repoUrl)).toPath();
 
-        // check if Repository object already created and available in cachedRepositories
-        if (cachedRepositories.containsKey(localPath)) {
-            // in this case we pull for changes to make sure the Git repo is up to date
-            Repository repository = cachedRepositories.get(localPath);
-            // disable auto garbage collection because it can lead to problems
-            repository.getConfig().setString("gc", null, "auto", "0");
-
-            pull(repository);
+        // First try to just retrieve the git repository from our server, as it might already be checked out.
+        try {
+            Repository repository = getRepositoryByLocalPath(localPath);
+            if (pullOnGet) {
+                pull(repository);
+            }
```

Comment:
Please add again the garbage collection setting. I added this recently due to a problem in the archive functionality. It might got lost during a merge conflict

Reply:
The garbage collection setting is still set, but only in the case of repository retrieval (getting a repository that is not already in the `cachedRepositories`): https://github.com/ls1intum/Artemis/pull/613/files#diff-5103a3e9a7d323e2f043351ac7a67022R173
When the repository is retrieved from `cachedRepositories`, it should still have the setting.
We could also add the line here, but it should be sufficient imo.